sdk/core/core-client-rest/test/getClient.spec.ts

```diff
@@ -6,16 +6,17 @@ import { getCachedDefaultHttpsClient } from "../src/clientHelpers";
 import { getClient } from "../src/getClient";
 import sinon from "sinon";
 import {
-  createHttpHeaders,
   PipelinePolicy,
   PipelineRequest,
   PipelineResponse,
   SendRequest,
+  createHttpHeaders,
 } from "@azure/core-rest-pipeline";
 
 describe("getClient", () => {
   afterEach(() => {
     sinon.reset();
+    sinon.restore();
```

Comment:
what's the difference between these? Maybe we have the wrong one in other places? 😅

Reply:
It looks like we need restore as we are working with the default workspace, restore seems to be the one used in most of our tests. I think 'reset' is useful at a stub level and restore at a workspace level